Explorer/Assets/DCL/CommunicationData/URLHelpers/URN.cs

```diff
@@ -137,13 +133,32 @@ public URN Shorten()
                     if (index == -1) break;
                 }
 
-                return index != -1 ? originalUrn[..index] : originalUrn;
+                return index != -1 ? GetShortenedUrn(index) : this;
             }
 
             // TokenId is always placed in the last part for regular nfts
             index = originalUrn.LastIndexOf(':');
 
-            return index != -1 ? originalUrn[..index] : this;
+            return index != -1 ? GetShortenedUrn(index) : this;
+        }
+
+        private URN GetShortenedUrn(int index)
+        {
+            if (SHORTENED_URNS_CACHE.TryGetValue(this, out URN shortenedUrn))
+                return shortenedUrn;
+
+            shortenedUrn = originalUrn[..index];
+
+            if (SHORTENED_URNS_CACHE.Count >= CACHE_MAX_SIZE)
```

Comment:
A better option would be some form of LRC cache this way we may be removing keys that are being used most often over and over again (assuming we hit the limit often)